python_modules/dagster/dagster/_core/definitions/run_request.py

```diff
@@ -276,6 +289,21 @@ def partition_key_range(self) -> Optional[PartitionKeyRange]:
         else:
             return None
 
+    def requires_backfill_daemon(self):
+        # TODO - is this slow?
+        return (
+            self.asset_graph_subset is not None
+            and len(
+                set(
+                    [
+                        asset_partition.partition_key
```

Comment:
yeah that's right -- particularly in cases where users are targeting all partitions of an asset with a bunch of partitions this can be quite slow.

At the very least, `AssetSubset.size` is often more efficient that `len(<list of all asset partitions>)`. So I think if AssetGraphSubset had a method for listing the internal `AssetSubsets` (right now it just has a `get_asset_subset()` method), then the logic could just look something like:

```python
any(
    asset_subset.size > 1 for asset_subset in self.asset_graph_subset.all_asset_subsets()
)
```

I think technically this is still not the exact right code, as depending on the backfill policy it's still possible to execute everything within a single run without the backfill daemon (i.e. in today's world a "single run backfill" doesn't touch the daemon). So there probably needs to be some extra logic to handle that case in the medium term.

Reply:
ok cool, I couldn't remember if the work on `AssetGraphSubset` was to make these iterations faster, or if it was to make them more obviously stand out as slow. I'll take another pass at this fn to make it more efficient